### ISSUE TITLE
How well do REPA-E perform if the tokenizer are randomly intialized?

### ISSUE THREAD

**Issue by yxchng** [End2End-Diffusion/REPA-E#9]
I have a question regarding the training process described in the paper associated with this repository.

The paper claims that the popular "two-stage division of the LDM training process leads to a challenging optimization task." It then proposes a method to "jointly tune both VAE and LDM in an end-to-end manner," supposedly achieving this in a single stage. However, if the VAE tokenizer is pre-trained, isn't the overall process still effectively two stages: first pre-training the tokenizer, and then jointly training the VAE and LDM? Do you have results for REPA-E when the tokenizer are randomly initialized?

Thank you.

**Comment by xingjianleng** [comment]
Hi thanks for the question.

We did the comparison between the REPA baseline, random initializing VAE weights, and initializing VAE from pretrained weights in **Table 8** of the paper. We found that while initializing the VAE with pretrained weights helps slightly improve performance, REPA-E can achieve significantly superior performance over REPA which requires a separate stage for training VAE.

**Comment by yxchng** [comment]
Ok I missed that. Do you have any intuition why random init is still weaker? Do you think we beat VAE init with random init in one stage?

**Comment by 1jsingh** [comment]
> Ok I missed that. Do you have any intuition why random init is still weaker? 

The performance difference between from scratch initialization and pretrained VAE initialization is actually quite small (4.34 vs 4.07 FID) as per Tab. 8, which might be because of the VAE prior providing a faster training in initial parts of training process (since VAE representation is already meaningful). 

> Do you think we beat VAE init with random init in one stage?

We currently only evaluate the from-scratch experiments until 80 epochs. With longer training it is possible that both methods actually converge to similar performance. We are still exploring the from-scratch experiments to see if it is possible to surpass VAE init with just single-stage end-to-end training.

**Comment by xmhGit** [comment]
> > Ok I missed that. Do you have any intuition why random init is still weaker?
> 
> The performance difference between from scratch initialization and pretrained VAE initialization is actually quite small (4.34 vs 4.07 FID) as per Tab. 8, which might be because of the VAE prior providing a faster training in initial parts of training process (since VAE representation is already meaningful).
> 
> > Do you think we beat VAE init with random init in one stage?
> 
> We currently only evaluate the from-scratch experiments until 80 epochs. With longer training it is possible that both methods actually converge to similar performance. We are still exploring the from-scratch experiments to see if it is possible to surpass VAE init with just single-stage end-to-end training.

Hi authors,

Thanks for your nice work. I'm wondering if the repo provide the code that allows us to train VAE with random initilization?
I just saw the code about loading pre-trained VAE models. We can just delete these pre-training parts to change to random initilization?
Thanks for your time!

Best,
Minghui

**Comment by xingjianleng** [comment]
Hi @xmhGit, thanks for your interest.

Yes, that’s the correct way — simply removing the pre-trained VAE loading will default to random initialization, and then you can run training from scratch.

Please feel free to reach out if you have any further questions. Thanks!

**Comment by xmhGit** [comment]
Hi Xingjian,

Thanks so much for your prompt reply and the interesting work.
I will try the model very soon!

Best,
Minghui

**Comment by Aryan-Satpathy** [comment]
Hi @xingjianleng,

I’m trying to train the REPA-E end-to-end from scratch. I commented out checkpoint loading and added explicit weight initialization for the VAE. However, generations collapse very quickly (noise → mostly black images within ~10 iterations).

### Setup
- **Model:** `SiT-XL/2`
- **Resolution:** `256`
- **Batch size:** `64`
- **Gradient accumulation steps:** `4`
- **Other hyperparameters:** defaults from `README.md`

### Questions
1. Are any hyperparameters expected to change when training E2E from scratch?
2. Should the pretrained discriminator still be used in this setting, or should it also be trained from scratch?
